src/utils/narrow.js

```diff
@@ -437,7 +438,18 @@ export const apiNarrowOfNarrow = (
       // TODO(server-2.1): just send IDs instead
       return [{ operator: 'pm-with', operand: emails.join(',') }];
     },
-    search: query => [{ operator: 'search', operand: query }],
+    // The search narrow can behave in two ways,
+    // one where there is no filters, so the user can search a message in all the streams
+    // the second where the user wants to search in a particular stream
+    // Below we are checking if the streamId is mentioned or not
+    // If it is mentioned, it will construct a narrow based on that.
```

Comment:
These two lines are an example of a comment that just repeats what the code already straightforwardly says. So it's better to leave them out and let the code speak for itself.

That's good because it means fewer things to read when seeking to understand the code. It also means one fewer thing that can get out of sync when the code changes, and become wrong -- as is, this sort of comment is highly vulnerable to becoming wrong in the future, and a comment that's wrong is of negative value.